src/core/include/openvino/opsets/factory.hpp

```diff
@@ -0,0 +1,67 @@
+// Copyright (C) 2018-2023 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
```

Comment:
I suppose we may make `factory.hpp` private for users and move out from `include/`

@ilya-lavrenov

Reply:
Is it used in test code as class also.
And also OpSet class use template function for insert (header only) and use this factory  template member.